### PR TITLE
importNet.MITAB

### PR DESCRIPTION
**Pull request by bronnil**
Not complete

### CODE REVIEW

File: R/importPSIMITAB.R [hunk 1]
```diff
@@ -0,0 +1,107 @@
+library(stringr)
+
+importNet.STRING <- function(fName,
```
Comment (RicardoRam) [inline R/importPSIMITAB.R]: add function definition

File: R/importPSIMITAB.R [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                             cutoffType = "xN",
+                             val,
+                             experimentType = getOptions("rete.defaultPPI"),
+                             taxID = "9606",
+                             silent = FALSE,
+                             noLog = FALSE) {
```
Comment (RicardoRam) [inline R/importPSIMITAB.R]: change variable noLog to writeLog

File: R/importPSIMITAB.R (hunk 1 above)
Comment (hyginn) [reply R/importPSIMITAB.R]: No ```library()``` call in function code. What do you even need ```stringr``` for?

File: R/importPSIMITAB.R [hunk 3]
```diff
@@ -0,0 +1,107 @@
+library(stringr)
+
```
Comment (RicardoRam) [inline R/importPSIMITAB.R]: Please paste R CMD check in a comment

File: R/importPSIMITAB.R [hunk 4]
```diff
@@ -0,0 +1,107 @@
+library(stringr)
```
Comment (RicardoRam) [inline R/importPSIMITAB.R]: remove library call

File: R/importPSIMITAB.R (hunk 2 above)
Comment (hyginn) [reply R/importPSIMITAB.R]: It seems you are working with really, really old code. This change in the ```importNet.STRING``` code was made long, long ago and if you don't start from the most recent code base, this is going to be a **painful** journey!

File: R/importPSIMITAB.R [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  } else {
+    # replace mapped IDs with their gene symbol
+    netStr$a[! is.na(mapA)] <- mapA[! is.na(mapA)]
+    netStr$b[! is.na(mapB)] <- mapB[! is.na(mapB)]
+  }
+  
+  gG <- .df2gG(netStr)
+  
+}
+
```
Comment (RicardoRam) [inline R/importPSIMITAB.R]: TODO: Add test cases and logs

File: R/importPSIMITAB.R (hunk 3 above)
Comment (hyginn) [reply R/importPSIMITAB.R]: @RicardoRam - I am encouraging everyone to pull-request early, so I/we can comment directly into code. FWIW they can paste R CMD check into the Issue when the code is really ready to by merged.

File: R/importPSIMITAB.R (hunk 2 above)
Comment (bronnil) [reply R/importPSIMITAB.R]: I was going with what was here http://steipe.biochemistry.utoronto.ca/abc/students/index.php/User:Boris/BCB420_2017_tasks/IMPORT-N.MITAB

I can change to writeLog?

File: R/importPSIMITAB.R (hunk 1 above)
Comment (bronnil) [reply R/importPSIMITAB.R]: Done

File: R/importPSIMITAB.R (hunk 2 above)
Comment (bronnil) [reply R/importPSIMITAB.R]: Done

File: R/importPSIMITAB.R (hunk 4 above)
Comment (bronnil) [reply R/importPSIMITAB.R]: Done

File: R/importPSIMITAB.R (hunk 5 above)
Comment (bronnil) [reply R/importPSIMITAB.R]: Working on it

File: R/importPSIMITAB.R (hunk 1 above)
Comment (bronnil) [reply R/importPSIMITAB.R]: I was using it to try to remove the "|" but forgot to remove it

File: R/importNet.MITAB.R [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    '.checkArgs> "writeLog" mode error')
+  
+  expect_false(file.exists(logName))
+
+}
+
+
+
+
+importNet.STRING <- function(fName,
```
Comment (RicardoRam) [inline R/importNet.MITAB.R:62]: function name is wrong

File: R/importNet.MITAB.R [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+                          cutoffType = "xN",
+                          silent = TRUE),
+    '.checkArgs> "fName' error: file "noSuch.txt does not exist.")
+  
+  # ==== dropUnmapped
+  
+  expect_error(
+    gG <- importNet.MITAB(fName = fN,
+                          cutffType = "xN",
+                          dropUnmapped = NULL,
```
Comment (RicardoRam) [inline R/importNet.MITAB.R:38]: function doesn't have arg "dropUnmapped"

File: R/importNet.MITAB.R [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    gG <- importNet.MITAB(fNAme = fN,
+                          cutoffType = "xN",
+                          silent = TRUE,
+                          writeLog = NULL),
+    '.checkArgs> "writeLog" mode error')
+  
+  expect_false(file.exists(logName))
+
+}
+
```
Comment (RicardoRam) [inline R/importNet.MITAB.R:58]: ToDo: add tests for cutoffType, val, experimentType, taxID

File: R/importNet.MITAB.R [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+    gG <- importNet.MITAB(fName = c("file_A", "file_B"),
+                          cutoffType = "xN",
+                          silent = TRUE),
+    '.checkArgs> "fName" length error')
+  
+  expect_error(
+    gG <- importNet.MITAB(fName = "noSuch.txt", 
+                          cutoffType = "xN",
+                          silent = TRUE),
+    '.checkArgs> "fName' error: file "noSuch.txt does not exist.")
```
Comment (RicardoRam) [inline R/importNet.MITAB.R:31]: string not formatted correctly

File: tests/testthat/testImportNet(Version2).MITAB.R [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+                          silent = TRUE),
+    '.checkArgs> "fName" length error')
+  
+  expect_error(
+    gG <- importNet.MITAB(fName = "noSuch.txt", 
+                          cutoffType = "xN",
+                          silent = TRUE),
+    '.checkArgs> "fName" error: file "noSuch.txt" does not exist.')
+  
+  # ==== dropUnmapped
```
Comment (RicardoRam) [inline tests/testthat/testImportNet(Version2).MITAB.R]: what is parameter dropUnmapped?

File: R/importNet.MITAB.R (hunk 7 above)
Comment (bronnil) [reply R/importNet.MITAB.R:38]: yeah it does its NULL

File: R/importNet.MITAB.R (hunk 9 above)
Comment (bronnil) [reply R/importNet.MITAB.R:31]: I dont understand what you mean?

File: tests/testthat/testImportNet(Version2).MITAB.R (hunk 10 above)
Comment (bronnil) [reply tests/testthat/testImportNet(Version2).MITAB.R]: TRUE or FALSE? Not sure what youre asking?

File: R/importNet.MITAB.R (hunk 8 above)
Comment (bronnil) [reply R/importNet.MITAB.R:58]: Those have tests in the code itself but ill add some here too then

File: R/importNet.MITAB.R (hunk 7 above)
Comment (RicardoRam) [reply R/importNet.MITAB.R:38]: Sorry, I wasn't clear enough. In the file  R/importPSIMITAB.R ,  the function signature doesn't have the parameter dropUnmapped.

File: tests/testthat/testImportNet(Version2).MITAB.R (hunk 10 above)
Comment (RicardoRam) [reply tests/testthat/testImportNet(Version2).MITAB.R]: Refer to your my other comment

File: R/importNet.MITAB.R (hunk 9 above)
Comment (hyginn) [reply R/importNet.MITAB.R:31]: Check your quotation marks

**Comment by hyginn** [comment]
You've actually got the MITAB code to a stage where it's actually going to be less work to revise your submission than rewrite from scratch. That's good I guess. Copied into the repo.

**Pull request closed without merging**